src/module_1/module_1_meteo_api.py

```diff
@@ -39,25 +50,28 @@ def call_api(url):
             logging.error(f"\nAPI request failed with Code: {response.status_code}")
             return None
     except requests.exceptions.RequestException as exception:
-            logging.error(f"\nAPI request failed with Exception: {exception}")
-            return None
+        logging.error(f"\nAPI request failed with Exception: {exception}")
+        return None
 
 
 def get_data_meteo_api(city, from_year, until_year):
     coordinates = COORDINATES.get(city)
     if coordinates is None:
         print("No data for the city")
         return None
-    
-    #Define local variables for lattitude and longitude
-    lat = coordinates['latitude']
-    long = coordinates['longitude']
 
-    #Create the final URL
-    url = f"{API_URL}latitude={lat}&longitude={long}&start_date={from_year}-01-01&end_date={until_year}-12-31&models={MODELS}&daily={VARIABLES}"
+    # Define local variables for lattitude and longitude
+    lat = coordinates["latitude"]
+    long = coordinates["longitude"]
 
-    #Define a num of max attempts for calling again
-    retry_count=0
+    # Create the final URL
+    url = (
+        f"{API_URL}latitude={lat}&longitude={long}&start_date={from_year}"
+        f"-01-01&end_date={until_year}-12-31&models={MODELS}&daily={VARIABLES}"
+    )
+
+    # Define a num of max attempts for calling again
+    retry_count = 0
 
     while retry_count < MAX_RETRY_ATTEMPTS:
```

Comment:
Solved both

Reply:
Additional question: If I run the code I obtain 3 times the print "INFO - API request connected successfully" as we're iterating through 
```
for city, coord in COORDINATES.items():
        data.append(
            pd.DataFrame(get_data_meteo_api(city, 1950, 2050)["daily"]).assign(
                city=city
            )
```
Is this the ideal expected behaviour (i.e. one success log per city request) or would be better idea just one-time log?

Reply:
it's not a big deal tbh. But ideally, parametrize the INFO in order the log the name of the city/url too so you get extra info at a very low cost, so: `logging.info(f"API request connected successfully to {url}")` or `logging.error(f"\nAPI request failed with Code: {response.status_code}" for url {ur})`. Otherwise is gonna be hard to know which `url` is failing